ProcessLib/SmallDeformationWithLIE/LocalAssembler/SmallDeformationLocalAssemblerMatrixNearFracture-impl.h

```diff
@@ -115,21 +118,60 @@ assembleWithJacobian(
 {
     assert (_element.getDimension() == DisplacementDim);
 
-    auto const n_dof_per_var = ShapeFunction::NPOINTS * DisplacementDim;
-
-    auto localRhs_ru = local_b.segment(0, n_dof_per_var);
-    auto localRhs_du = local_b.segment(n_dof_per_var, n_dof_per_var);
-
-    auto localA_uu = local_J.block(0, 0, n_dof_per_var, n_dof_per_var);
-    auto localA_ug = local_J.block(0, n_dof_per_var, n_dof_per_var, n_dof_per_var);
-    auto localA_gu = local_J.block(n_dof_per_var, 0, n_dof_per_var, n_dof_per_var);
-    auto localA_gg = local_J.block(n_dof_per_var, n_dof_per_var, n_dof_per_var, n_dof_per_var);
-
-    auto const nodal_ru = local_u.segment(0, n_dof_per_var);
-    auto nodal_du = local_u.segment(n_dof_per_var, n_dof_per_var);
+    auto const N_DOF_PER_VAR = ShapeFunction::NPOINTS * DisplacementDim;
+    auto const n_fractures = _fracture_props.size();
+
+    using BlockVectorType = typename Eigen::VectorXd::FixedSegmentReturnType<N_DOF_PER_VAR>::Type;
+    using BlockMatrixType = Eigen::Block<Eigen::MatrixXd,N_DOF_PER_VAR,N_DOF_PER_VAR>;
+
+    //--------------------------------------------------------------------------------------
+    // prepare sub vectors, matrices for regular displacement (u) and displacement jumps (g)
+    //
+    // example with two fractures:
+    //     |b(u)|
+    // b = |b(g1)|
+    //     |b(g2)|
+    //
+    //     |J(u,u)  J(u,g1)  J(u,g2) |
+    // J = |J(g1,u) J(g1,g1) J(g1,g2)|
+    //     |J(g2,u) J(g2,g1) J(g2,g2)|
+    //--------------------------------------------------------------------------------------
+    auto local_b_u = local_b.segment<N_DOF_PER_VAR>(0);
+    std::vector<BlockVectorType> vec_local_b_g;
+    for (unsigned i=0; i<n_fractures; i++)
+        vec_local_b_g.push_back(local_b.segment<N_DOF_PER_VAR>(N_DOF_PER_VAR*(i+1)));
+
+    auto local_J_uu = local_J.block<N_DOF_PER_VAR, N_DOF_PER_VAR>(0, 0);
+    std::vector<BlockMatrixType> vec_local_J_ug;
+    std::vector<BlockMatrixType> vec_local_J_gu;
+    std::vector<std::vector<BlockMatrixType>> vec_local_J_gg(n_fractures);
```

Comment:
Since this code is executed in every local assemble, I think it would be very good to do the reservation.
It's three lines of code. Please do it. We do it in many other places, too. It stresses good style.